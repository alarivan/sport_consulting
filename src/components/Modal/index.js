import React, { useEffect } from "react"

import {
  ModalBackground,
  ModalContainer,
  ModalHeader,
  ModalHeaderTitle,
  ModalHeaderClose,
} from "./styles"
import IconCross from "../icons/Cross"
import { useSpring, animated } from "react-spring"

const AnimatedModalContainer = animated(ModalContainer)

export default function Modal({ isOpen, onClose, title, children }) {
  const contentProps = useSpring({
    opacity: isOpen ? 1 : 0,
    config: { duration: 100 },
  })

  useEffect(() => {
    window.addEventListener("keydown", onEscKeyDown, false)

    return () => {
      window.removeEventListener("keydown", onEscKeyDown, false)
    }
  }, [isOpen])

  const onEscKeyDown = e => {
    if (e.key !== "Escape") return

    onClose()
  }

  if (isOpen) {
    return (
      <ModalBackground open={isOpen} onClick={onClose}>
        <AnimatedModalContainer
          style={contentProps}
          onClick={event => event.stopPropagation()}
        >
          <ModalHeader>
            <ModalHeaderTitle>{title || ""}</ModalHeaderTitle>
            <ModalHeaderClose onClick={onClose}>
              <IconCross />
            </ModalHeaderClose>
          </ModalHeader>
          {children}
        </AnimatedModalContainer>
      </ModalBackground>
    )
  } else {
    typeof document !== `undefined` &&
      document.body.style.removeProperty(`overflow-y`)
    return null
  }
}
