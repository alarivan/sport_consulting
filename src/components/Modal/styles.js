import styled from "styled-components"

export const ModalBackground = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
  display: grid;
  visibility: ${props => (props.open ? `visible` : `hidden`)};
  opacity: ${props => (props.open ? `1` : `0`)};
  transition: 0.5s;
  z-index: 2;
`

export const ModalContainer = styled.div`
  max-height: 100%;
  background: #fff;
  box-sizing: border-box;
  align-self: center;
  justify-self: center;
  max-width: 480px;
  padding: 1rem;
  width: 100%;
  position: relative;
  overflow: auto;
  transition: opacity 400ms ease-in-out;
  box-shadow: 0 0 3rem black;
  margin: calc(0.5rem + 2vw);
`

export const ModalHeader = styled.header`
  padding: 0 0 0.5rem 0;
  margin-bottom: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  border-bottom: 1px solid ${props => props.theme.borderColor};
`

export const ModalHeaderTitle = styled.h1`
  flex: 1 1 auto;
  font-size: 1.5rem;
  font-weight: 400;
  margin: 0 0.5rem 0 0;
`

export const ModalHeaderClose = styled.button`
  background: none;
  border: none;
  cursor: pointer;
  outline: none;

  &:hover {
    opacity: 0.7;
  }
`
