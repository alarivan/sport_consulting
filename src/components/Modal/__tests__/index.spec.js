import React from "react"
import { render, fireEvent } from "@testing-library/react"
import "jest-styled-components"
import Modal from "../index"

describe("Modal/index", () => {
  it("renders correctly", () => {
    const props = {
      onClose: jest.fn(),
      isOpen: true,
      title: "Title",
    }
    const { asFragment, queryByText } = render(
      <Modal {...props}>Content</Modal>
    )

    expect(queryByText(/Title/i)).not.toBeNull()
    expect(queryByText(/Content/i)).not.toBeNull()

    expect(asFragment()).toMatchSnapshot()
  })

  it("calls onClose", () => {
    const props = {
      onClose: jest.fn(),
      isOpen: true,
      title: "Title",
    }
    const { getByTestId } = render(<Modal {...props}>Content</Modal>)

    fireEvent.click(getByTestId("modal-header-close"))
    expect(props.onClose).toHaveBeenCalledTimes(1)

    fireEvent.click(getByTestId("modal-background"))
    expect(props.onClose).toHaveBeenCalledTimes(2)
  })
})
