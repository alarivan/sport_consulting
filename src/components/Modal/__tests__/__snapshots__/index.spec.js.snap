// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal/index renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.5);
  display: grid;
  -webkit-transition: 0.5s;
  transition: 0.5s;
  z-index: 2;
}

.c1 {
  max-height: 100%;
  background: #fff;
  box-sizing: border-box;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: center;
  max-width: 480px;
  padding: 1rem;
  width: 100%;
  position: relative;
  overflow: auto;
  -webkit-transition: opacity 400ms ease-in-out;
  transition: opacity 400ms ease-in-out;
  box-shadow: 0 0 3rem black;
  margin: calc(0.5rem + 2vw);
}

.c2 {
  padding: 0 0 0.5rem 0;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid;
}

.c3 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: 1.5rem;
  font-weight: 400;
  margin: 0 0.5rem 0 0;
}

.c4 {
  background: none;
  border: none;
  cursor: pointer;
  outline: none;
}

.c4:hover {
  opacity: 0.7;
}

.c5 {
  display: block;
  width: 1rem;
  height: 1rem;
}

<div
    class="c0"
    data-testid="modal-background"
    open=""
  >
    <div
      class="c1"
      style="opacity: 1;"
    >
      <header
        class="c2"
      >
        <h1
          class="c3"
        >
          Title
        </h1>
        <button
          class="c4"
          data-testid="modal-header-close"
        >
          <svg
            class="c5"
            viewBox="0 0 32 32"
          >
            <path
              d="M31.708 25.708c-0-0-0-0-0-0l-9.708-9.708 9.708-9.708c0-0 0-0 0-0 0.105-0.105 0.18-0.227 0.229-0.357 0.133-0.356 0.057-0.771-0.229-1.057l-4.586-4.586c-0.286-0.286-0.702-0.361-1.057-0.229-0.13 0.048-0.252 0.124-0.357 0.228 0 0-0 0-0 0l-9.708 9.708-9.708-9.708c-0-0-0-0-0-0-0.105-0.104-0.227-0.18-0.357-0.228-0.356-0.133-0.771-0.057-1.057 0.229l-4.586 4.586c-0.286 0.286-0.361 0.702-0.229 1.057 0.049 0.13 0.124 0.252 0.229 0.357 0 0 0 0 0 0l9.708 9.708-9.708 9.708c-0 0-0 0-0 0-0.104 0.105-0.18 0.227-0.229 0.357-0.133 0.355-0.057 0.771 0.229 1.057l4.586 4.586c0.286 0.286 0.702 0.361 1.057 0.229 0.13-0.049 0.252-0.124 0.357-0.229 0-0 0-0 0-0l9.708-9.708 9.708 9.708c0 0 0 0 0 0 0.105 0.105 0.227 0.18 0.357 0.229 0.356 0.133 0.771 0.057 1.057-0.229l4.586-4.586c0.286-0.286 0.362-0.702 0.229-1.057-0.049-0.13-0.124-0.252-0.229-0.357z"
            />
          </svg>
        </button>
      </header>
      Content
    </div>
  </div>
</DocumentFragment>
`;
